app/domain/authentication/authn_jwt/input_validation/parse_mapped_claims.rb

```diff
@@ -0,0 +1,73 @@
+module Authentication
+  module AuthnJwt
+    module InputValidation
+      # Parse mapped-claims secret value and return a validated claims list
+      ParseMappedClaims ||= CommandClass.new(
+        dependencies: {
+          validate_claim_name: ValidateClaimName.new(
+            deny_claims_list_value: MANDATORY_CLAIMS_DENY_LIST
+          ),
+          logger: Rails.logger
+        },
+        inputs: %i[mapped_claims_secret_value]
+      ) do
+        def call
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ParsingMappedClaims.new(@mapped_claims_secret_value))
+          validate_mapped_claims_secret_value_exists
+          validate_mapped_claims_list_values
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ParsingMappedClaims.new(mapping_hash))
+          mapping_hash
+        end
+
+        private
+
+        def validate_mapped_claims_secret_value_exists
+          raise Errors::Authentication::AuthnJwt::MappedClaimsMissingInput if @mapped_claims_secret_value.blank?
+        end
+
+        def validate_mapped_claims_list_values
+
+          raise Errors::Authentication::AuthnJwt::MappedClaimsInvalidFormat, @mapped_claims_secret_value if
+            mapping_tuples_list.empty?
+
+          mapping_tuples_list.each do |tuple|
+            from, to = mapping_tuple(tuple)
```

Comment:
@tzheleznyak @sashaCher @hilagross @nessiLahav let's take a strict decision. 
We've talked about "new name:claim name in token​" , here its "claim name in token​:new name" if i understand it correctly.. 
I'm fine with both but lets make sure we are aligned

Reply:
Done. annotation name:claim name